main/manager.c

```diff
@@ -6873,6 +6900,203 @@ static int action_coreshowchannels(struct mansession *s, const struct message *m
 	return 0;
 }
 
+/*! \brief Destructor for channel map vector */
+static void channel_map_vector_destructor(void *obj)
+{
+	struct ast_vector_string *channel_map = obj;
+	AST_VECTOR_RESET(channel_map, ast_free);
+	AST_VECTOR_FREE(channel_map);
+}
+
+/*! \brief Helper function to add a channel name to the vector */
+static int coreshowchannelmap_add_to_map(struct ast_vector_string *v, const char *s)
+{
+	char *str;
+	int size;
+
+	/* If we've already added this channel, don't add it again */
+	if (AST_VECTOR_GET_CMP(v, s, !strcmp)) {
+                return -1;
+        }
+
+	size = strlen(s) + 1;
```

Comment:
I'm probably missing something here, but what's the reason for using malloc + ast_copy_string rather than ast_strdup?

Also, if the allocation succeeds, there is no need to bounds check. strcpy would be safe.